implementation/clabbert/src/main/java/pwr/jakprzyjade/clabbert/api/configurations/ApiConfig.java

```diff
@@ -0,0 +1,22 @@
+/* @jakubzehner (C) 2024 */
+package pwr.jakprzyjade.clabbert.api.configurations;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.lang.NonNull;
+import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+import pwr.jakprzyjade.clabbert.api.middlewares.UserHeaderInterceptor;
+
+@Configuration
+@RequiredArgsConstructor
+public class ApiConfig implements WebMvcConfigurer {
+
+    @Autowired private final UserHeaderInterceptor userHeaderInterceptor;
```

Comment:
**suggestion:** It is commonly believed in that in newer Spring versions, constructor injection should be preferred to field injection:
- https://odrotbohm.de/2013/11/why-field-injection-is-evil/
- https://stackoverflow.com/questions/40620000/spring-autowired-on-properties-vs-constructor
- https://reflectoring.io/constructor-injection/

You even used `@RequiredArgsConstructor`, is there any reason to add the `@Autowired`?

Reply:
Full agreement, that's what I wanted to do too, `@Autowired` stayed there by accident. Thanks for catching that 👍